hack/e2e-common.sh

```diff
@@ -0,0 +1,181 @@
+max_tries=60
+sleep_between_tries=10
+# Set timeout for the cluster deployment to install
+# timeout = sleep_between_cluster_deployment_status_checks * max_cluster_deployment_status_checks
+max_cluster_deployment_status_checks=90
+sleep_between_cluster_deployment_status_checks="1m"
+
+export CLUSTER_NAMESPACE="${CLUSTER_NAMESPACE:-cluster-test}"
+
+# In CI, HIVE_IMAGE and RELEASE_IMAGE are set via the job's `dependencies`.
+if [[ -z "$HIVE_IMAGE" ]]; then
+    echo "The HIVE_IMAGE environment variable was not found." >&2
+    echo "It must be set to the fully-qualified pull spec of a hive container image." >&2
+    echo "E.g. quay.io/my-user/hive:latest" >&2
+    exit 1
+fi
+if [[ -z "$RELEASE_IMAGE" ]]; then
+    echo "The RELEASE_IMAGE environment variable was not found." >&2
+    echo "It must be set to the fully-qualified pull spec of an OCP release container image." >&2
+    echo "E.g. quay.io/openshift-release-dev/ocp-release:4.7.0-x86_64" >&2
+    exit 1
+fi
+
+echo "Running ${TEST_NAME} with HIVE_IMAGE ${HIVE_IMAGE}"
+echo "Running ${TEST_NAME} with RELEASE_IMAGE ${RELEASE_IMAGE}"
+
+i=1
+while [ $i -le ${max_tries} ]; do
+  if [ $i -gt 1 ]; then
+    # Don't sleep on first loop
+    echo "sleeping ${sleep_between_tries} seconds"
+    sleep ${sleep_between_tries}
+  fi
+
+  echo -n "Creating namespace ${CLUSTER_NAMESPACE}. Try #${i}/${max_tries}... "
+  if oc create namespace "${CLUSTER_NAMESPACE}"; then
+    echo "Success"
+    break
+  else
+    echo -n "Failed, "
+  fi
+
+  i=$((i + 1))
+done
```

Comment:
should we not need to verify that the namespace creation was successfull when we exit this loop (so we can have a clearer error message in the even that this never succeeded)?

Reply:
Absolutely. There's a couple of things here, including the fact that the command to restore the original namespace doesn't work at all for me. I'd like to defer that to a separate PR if possible.